wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 KURZ Digital Solutions GmbH
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#import "ZXIBarcodeReader.h"
+#import "ZXIErrors.h"
+#import "ZXing/ReadBarcode.h"
+#import "ZXing/ImageView.h"
+#import "ZXing/Result.h"
+#import "ZXIFormatHelper.h"
+
+using namespace ZXing;
+
+@interface ZXIBarcodeReader()
+@property (nonatomic, strong) CIContext* ciContext;
+@end
+
+@implementation ZXIBarcodeReader
+
+- (instancetype)init {
+    return [self initWithHints: [[ZXIDecodeHints alloc]initWithTryHarder:NO tryRotate:NO formats:@[]]];
+}
+
+- (instancetype)initWithHints:(ZXIDecodeHints*)hints{
+    self = [super init];
+    self.ciContext = [CIContext new];
+    self.hints = hints;
+    return self;
+}
+
+- (nullable ZXIResult *)readCIImage:(nonnull CIImage *)image error:(NSError **)error {
+    CGImageRef cgImage = [self.ciContext createCGImage:image fromRect:image.extent];
+    ZXIResult *result = [self readCGImage:cgImage error:error];
+    CGImageRelease(cgImage);
+    return result;
+}
+
+- (nullable ZXIResult *)readCGImage: (nonnull CGImageRef)image error:(NSError **)error {
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericGray);
+    CGFloat cols = CGImageGetWidth(image);
+    CGFloat rows = CGImageGetHeight(image);
+    NSMutableData *data = [NSMutableData dataWithLength: cols * rows];
+
+    CGContextRef contextRef = CGBitmapContextCreate(
```

Comment:
I don't know about the iOS world here, but I assume the 'native' formats of the cameras are similar compared to the android world, i.e. the most common format would be some type of YUV pixel forma. In that case there would be no image conversion necessary and simply a waste of computing resources, because all we need is the Y-plane. Please also look at [the qt example](https://github.com/nu-book/zxing-cpp/blob/bee813aebf01d42476b1b5c463f9e07ed0067443/example/ZXingQtReader.h#L165) for reference.

Reply:
I've looked into this but it's not as straight forward as expected. The [colorspace](https://developer.apple.com/documentation/coregraphics/cgcolorspace/color_space_names?changes=latest_minor) property of a `CGImageRef` does not even have the option to be YUV. The YUV colorspace is only available for pixel buffers which are obtained directly from the capture session. So I think we would have to create a designated API that deals with camera pixel buffers and not `CGImages`. I have to research a bit more on this topic.

Reply:
Thanks for looking into that. It might very well be worth it in terms of performance in the long-run. If I profile a standard situation with a single QRCode symbol and process RGB data, then a significant amount of time is spent on the simple RGB to Greyscale conversion (over 30% IIRC), which completely goes away if you can input the `Y` channel directly.

Reply:
Nice! And that is already used in the demo app? Did you notice any performance difference?

Reply:
Yes, we tested it with a M1 Mac simulator. The direct passing of the luminance channel seems to be much faster! To make a more accurate statement, we would have to test it on a device. 
The demo app already uses the new interface.

Reply:
Very nice. (I only saw your feedback now...)